### DISCUSSION TITLE
[Thesis] Blockly/Interpreter: Dungeon als Laufzeitumgebung

### DISCUSSION THREAD

**Discussion by cagix** [Programmiermethoden/Dungeon#1389] [Ideas]
In #1066 wurde ein erster Draft für die Integration von Blockly in den Dungeon gemacht. Das sollte ausgebaut werden, so dass man damit Erstis Programmieren und Algorithmen und Datenstrukturen beibringen kann. 

Letztlich wird damit unsere DSL durch Blockly als weiteres Frontend ergänzt (Use-Cases: Einsatz für Schüler wie in #645, Formulierung einfacher Aufgaben durch Lehrende), und es werden fest kodierte Spielsituationen eingesetzt. Der Dungeon wird als Laufzeitumgebung betrachtet.

- [ ] Definition geeigneter Aufgaben (@bcg7)
- [ ] Identifikation fehlender Konzepte (Blockly, Dungeon)
- [ ] Definition von geeigneten Spielsituationen 
- [ ] Umsetzung ... 

Das könnte ein Teil der Aufgaben für eine Praxisphase werden oder auch für eine Semester-/Abschlussarbeit.

Edit: Idee aus https://github.com/Programmiermethoden/2022-freiraum-dungeon/pull/239#discussion_r1346189386: @Lena241 und Finn könnten gemeinsam im November volle Kraft das Blockly-Problem lösen und Finn anschließend ab Mitte Dezember und im Januar die HUD-Sache.

siehe auch #645

**Comment by AMatutat** [comment]
Ich zieh mich hier mal raus, da das streng genommen nicht mein Bier ist. Bei einem Brainstorm Meeting wäre ich natürlich gerne dabei
